src/utils/sortedData.js

```diff
@@ -0,0 +1,35 @@
+import { comparePurchaseUrgency } from '../api';
+
+export function filteredData(data, searchString) {
+	return data.filter((d) =>
+		d.name?.toLowerCase().includes(searchString.toLowerCase()),
+	);
+}
+
+export function sortedData(filteredDataResult) {
+	return comparePurchaseUrgency(filteredDataResult);
+}
+
+export function sortedItems(sortedDataResult) {
+	const overdue = [];
+	const buySoon = [];
+	const buyKindOfSoon = [];
+	const buyNotSoon = [];
+	const inactive = [];
+
+	sortedDataResult.forEach((d) => {
+		if (d.daysUntilNextPurchase <= 0) {
+			overdue.push(d);
+		} else if (d.daysUntilNextPurchase <= 3) {
+			buySoon.push(d);
+		} else if (d.daysUntilNextPurchase <= 7) {
+			buyKindOfSoon.push(d);
+		} else if (d.daysUntilNextPurchase <= 14) {
+			buyNotSoon.push(d);
+		} else {
+			inactive.push(d);
+		}
+	});
```

Comment:
The conditionals in the forEach seem to be little off from the original for each urgency category, which I think led to the displayed data being shifted in sorting below: 
<img width="416" alt="Screen Shot 2024-03-29 at 5 05 59 PM" src="https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/64502078/26ee7f2a-b52a-4e2e-bb05-61e697bb9e55">

If helpful, here are the original conditionals: 
```sortedData.forEach((item) => {
		if (item.isOverdue) {
			overdue.push(item);
		} else if (item.daysSinceLastPurchase >= 60) {
			inactive.push(item);
		} else if (item.daysUntilNextPurchase <= 7) {
			buySoon.push(item);
		} else if (
			item.daysUntilNextPurchase > 7 &&
			item.daysUntilNextPurchase < 15
		) {
			buyKindOfSoon.push(item);
		} else if (item.daysUntilNextPurchase >= 15) {
			buyNotSoon.push(item);
		}
	});```

Reply:
Updated that section with your suggestions. Nice catch, thanks for pointing that out.